File: proje1/Akademisyen.java
public class Akademisyen {
    private String ad;
    private String soyad;

    public Akademisyen(String ad,String soyad){
        this.ad=ad;
        this.soyad=soyad;
    }
    public String getAd(){
        return ad;
    }
    public String getSoyad(){
        return soyad;
    }   
    public void notGir(Ders ders, ogrenci ogrenci, double... notlar) {
        if (ders.ogrenciKayitliMi(ogrenci.getSchoolNumber())) {
            ders.notGir(this, ogrenci, notlar);
        } else {
            System.out.println("Öğrenci bu derse kayitli değil.");
        }
    }
    
}

File: proje1python/Akademisyen.py
class Akademisyen:
    def __init__(self, ad, soyad):
        self.ad = ad
        self.soyad = soyad

    def get_ad(self):
        return self.ad

    def get_soyad(self):
        return self.soyad

    def not_gir(self, ders, ogrenci, *notlar):
        if ders.ogrenci_kayitli_mi(ogrenci.get_school_number()):
            ders.not_gir(self, ogrenci, *notlar)
        else:
            print("Öğrenci bu derse kayıtlı değil.")
